tests/core/channels/test_twilio_voice.py

```diff
@@ -0,0 +1,206 @@
+        "initial_prompt": "hello",
+        "reprompt_fallback_phrase": "i didn't get that",
+        "speech_model": "default",
+        "speech_timeout": "5",
+        "assistant_voice": "woman",
+        "enhanced": "false"
+    }
+
+    tv = TwilioVoiceInput(**inputs)
+
+    output_channel = TwilioVoiceCollectingOutputChannel()
+    await output_channel.send_text_with_buttons(
+        recipient_id="Chuck Norris",
+        text="Buttons:",
+        buttons=[
+            {"title": "Yes", "payload": "/affirm"},
+            {"title": "No", "payload": "/deny"},
+        ],
+    )
+    assert len(output_channel.messages) == 3
+    message_str = " ".join([m["text"] for m in output_channel.messages])
+    assert message_str == "Buttons: Yes No"
+
+    twiml = tv.build_twilio_voice_response(output_channel.messages)
+    assert (
+        str(twiml)
+        == '<?xml version="1.0" encoding="UTF-8"?><Response><Say voice="woman">Buttons:</Say><Pause length="1" /><Say voice="woman">Yes</Say><Pause length="1" /><Gather action="/webhooks/twilio_voice/webhook" actionOnEmptyResult="true" enhanced="false" input="speech" speechModel="default" speechTimeout="5"><Say voice="woman">No</Say></Gather></Response>'
+    )
+
+
+@pytest.mark.parametrize(
+    "configs, expected",
+    [
+        (
+            {
+                "initial_prompt": "hello",
+                "reprompt_fallback_phrase": "i didn't get that",
+                "speech_model": "default",
+                "speech_timeout": "5",
+                "assistant_voice": "alien",
+                "enhanced": "false"
+            }, InvalidConfigException
+        ),
+        (
+            {
+                "initial_prompt": "hello",
+                "reprompt_fallback_phrase": "i didn't get that",
+                "speech_model": "default",
+                "speech_timeout": "not a number",
+                "assistant_voice": "woman",
+                "enhanced": "false"
+            }, InvalidConfigException
+        ),
+        (
+            {
+                "initial_prompt": "hello",
+                "reprompt_fallback_phrase": "i didn't get that",
+                "speech_model": "default",
+                "speech_timeout": "auto",
+                "assistant_voice": "woman",
+                "enhanced": "wrong"
+            }, InvalidConfigException
+        ),
+        (
+            {
+                "initial_prompt": "hello",
+                "reprompt_fallback_phrase": "i didn't get that",
+                "speech_model": "default",
+                "speech_timeout": "5",
+                "assistant_voice": "woman",
+                "enhanced": "true"
+            }, InvalidConfigException
+        )
+    ]
+)
+def test_invalid_configs(configs, expected):
+    with pytest.raises(expected):
+        TwilioVoiceInput(**configs)
+
+
+async def test_twilio_voice_remove_image():
+
+    with pytest.warns(UserWarning):
+        output_channel = TwilioVoiceCollectingOutputChannel()
```

Comment:
optional: for clarity I'd only include the call with raises the warning in the `with pytest.warns` block. Otherwise it's hard to see which part of the code is exactly doing this.

Reply:
Good idea. I'm separating this into two tests. One to make sure the warning gets raised and another to make sure the image is removed and the text is passed through.